### PR TITLE
Schema, model, service, and test for TagLink

### PR DESCRIPTION
**Pull request by bzwei**

### CODE REVIEW

File: app/services/workflow_link_service.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    nil
+  end
+
+  def unlink(options)
+    TagLink.find_by(tag_link(options)).try(:destroy)
+  end
+
+  def find(options_arr)
+    options_arr.collect do |options|
+      TagLink.where(tag_link(options)).pluck(:workflow_id).first
```
Comment (mkanoor) [inline app/services/workflow_link_service.rb]: @bzwei Is this returning an array of workflow ids? How would we know which array entry corresponds to which workflow.

File: app/services/workflow_link_service.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    end
+  end
+
+  private
+
+  def tag_link(options)
+    options.merge(:workflow_id => workflow_id)
```
Comment (mkanoor) [inline app/services/workflow_link_service.rb]: @bzwei Why is this merging the workflow_id, shouldn't it be using the object_type, app_name to do the match and then pickup the TagLink object.

File: app/services/workflow_link_service.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+class WorkflowLinkService
+  attr_accessor :workflow_id
+
+  def initialize(workflow_id)
+    self.workflow_id = workflow_id
+  end
+
+  def link(options)
+    TagLink.where(tag_link(options)).first_or_create
+    nil
```
Comment (hsong-rh) [inline app/services/workflow_link_service.rb]: Why we force to return `nil` here?

File: app/services/workflow_link_service.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+class WorkflowLinkService
+  attr_accessor :workflow_id
+
+  def initialize(workflow_id)
+    self.workflow_id = workflow_id
+  end
+
+  def link(options)
```
Comment (hsong-rh) [inline app/services/workflow_link_service.rb]: Feel it's more straightforward to use `tags` than `options`, as parameter's name. Same thing for other methods.

File: app/services/workflow_link_service.rb (hunk 1 above)
Comment (bzwei) [reply app/services/workflow_link_service.rb]: This is like an array transform method many languages support: the output array has the same  length of the input, one to one mapping.

File: app/services/workflow_link_service.rb (hunk 3 above)
Comment (bzwei) [reply app/services/workflow_link_service.rb]: The caller only needs to know whether it succeeds or not. No `TagLink` object is required.

**Pull request merged**